native/src/main/java/io/ballerina/stdlib/oauth2/OAuth2Client.java

```diff
@@ -301,6 +299,9 @@ private static Object callEndpoint(HttpClient client, HttpRequest request) {
             return createError("Failed to get a success response from the endpoint. Response code: '" +
                                        response.statusCode() + "'. Response body: '" + response.body() + "'");
         } catch (IOException | InterruptedException e) {
+            if (e instanceof InterruptedException) {
+                Thread.currentThread().interrupt(); // Restore interrupted status
+            }
```

Comment:
Here we have already handled the InterruptedException by throwing an error to the Ballerina side via the `createError` method. It's not necessary to restore the status here. Is there a any particular reason to introduce this change ?

Reply:
I added this just for additional security measures and to handle errors more gracefully

Reply:
Let's remove this as there's no need to propagate this interruption.